modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/expression/MBColor.java

```diff
@@ -0,0 +1,99 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2018, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+
+package org.geotools.mbstyle.expression;
+
+import org.geotools.mbstyle.parse.MBFormatException;
+import org.geotools.mbstyle.parse.MBObjectParser;
+import org.json.simple.JSONArray;
+import org.opengis.filter.expression.Expression;
+import org.opengis.filter.expression.Function;
+
+/**
+ * A class to transform mapbox color expressions into geotools expressions.
+ */
+public class MBColor extends MBExpression{
+    MBObjectParser parse;
+
+    public MBColor(JSONArray json) {
+        super(json);
+        parse = new MBObjectParser(MBExpression.class);
+    }
+    /**
+     * Creates a color value from red, green, and blue components, which must range between 0 and 255,
+     * and an alpha component of 1. If any component is out of range, the expression is an error.
+     * Example: ["rgb", number, number, number]: color
+     * @return
+     */
+    public Expression colorRGB() {
+        if (json.size() == 4) {
+            Expression er = parse.string(json, 1);
+            er = transformLiteral(er);
+            Function roundr = ff.function("round_2", er);
+
+            Expression eg = parse.string(json, 2);
+            eg = transformLiteral(eg);
+            Function roundg = ff.function("round_2", eg);
+
+
+            Expression eb = parse.string(json, 3);
+            eb = transformLiteral(eb);
+            Function roundb = ff.function("round_2", eb);
+
+            Function color = ff.function("torgb", roundr, roundg, roundb);
+            return color;
+        } else {
+            throw new MBFormatException("Expression \"rgb\" requires exactly 3 arguments");
+        }
+    }
+
+    /**
+     * Creates a color value from red, green, blue components, which must range between 0 and 255, and an alpha
+     * component which must range between 0 and 1. If any component is out of range, the expression is an error.
+     * Example: ["rgba", number, number, number, number]: color
+     * @return
+     */
+    // Currently unsupported
+    public Expression colorRGBA(){
+        throw new UnsupportedOperationException("RGBA colors are currently unsupported");
```

Comment:
May be able to support by generating the colour, and generating the opacity.

We have a few other examples where a single Mapbox style option is split up across several geotools style attributes.